bread/static/js/main.js

```diff
@@ -175,3 +166,16 @@ function submitbulkaction(table, actionurl, method="GET") {
     document.body.appendChild(form);
     form.submit();
 }
+
+// helper functions to set and get bread-namespaced cookies
+function setBreadCookie(key, value) {
+    document.cookie = "bread-" + key + "=" + encodeURIComponent(value) + "; path=/";
+}
+
+function getBreadCookie(key, _default=null) {
+    var ret = document.cookie.split('; ').find(row => row.startsWith("bread-" + key + '='))
+    if(!ret)
+        return _default;
+    ret = ret.split('=')[1];
+    return ret ? decodeURIComponent(ret) : _default;
+}
```

Comment:
These cookie functions can be used to communicate client side changes which affect the generation of the user interface on the server side.